### PR TITLE
CI requirements update (macOS, GeoClaw netCDF)

### PR DESCRIPTION
**Pull request by scivision**
ci: macos: update GCC version. 

GeoClaw now requires netCDF.

Simplify CMake presets.
    
GCC-11 is being removed from GitHub Actions macOS runners in August 2024.
    
https://github.com/actions/runner-images/issues/10213

### CODE REVIEW

**Comment by donnaaboise** [comment]
I would prefer if GeoClaw didn't require NETCDF.    We rarely use it and so do not  require it.

**Comment by donnaaboise** [comment]
NETCDF shouldn't be required for GeoClaw.  We rarely use it and so do not require it.

**Comment by scottaiton** [comment]
Netcdf is not a required library at all. Things should be able to work without it.

**Comment by scottaiton** [comment]
The problem is probably related to geoclaw needing preprocessing for the fortran source files

File: CMakePresets.json [hunk 1]
```diff
@@ -1,16 +1,10 @@
...
-    "minor": 20,
-    "patch": 0
-},
+  "version": 4,
 
 "configurePresets": [
 {
   "name": "ci",
   "binaryDir": "${sourceDir}/build",
-  "generator": "Ninja",
```
Comment (scottaiton) [inline CMakePresets.json:13]: Please stop removing Ninja as the ci generator. We need it for Geoclaw since it makes heavy use of f90 modules

**Comment by scottaiton** [comment]
Closing, this has been implemented with pull requests #365 (Have CMake preprocess all fortran source files) and #361 (Use gcc-14 in macOS ci)

**Pull request closed without merging**